src/v0/destinations/fb_custom_audience/util.js

```diff
@@ -0,0 +1,187 @@
+const sha256 = require('sha256');
+const get = require('get-value');
+const jsonSize = require('json-size');
+
+const {
+    isDefinedAndNotNull,
+    returnArrayOfSubarrays
+} = require('../../util');
+const { maxPayloadSize } = require("./config");
+const {
+    InstrumentationError,
+    ConfigurationError,
+} = require('../../util/errorTypes');
+
+/**
+ * Example payload ={
+            "is_raw": true,
+            "data_source": {
+              "sub_type": "ANYTHING"
+            },
+            "schema": [
+              "EMAIL",
+              "COUNTRY"
+            ],
+            "data": [
+              [
+                "shrouti@abc.com",
+                "IN"
+              ]
+            ]
+} */
+const batchingWithPayloadSize = (payload) => {
+    const payloadSize = jsonSize(payload);
+    if (payloadSize > maxPayloadSize) {
+        const revisedPayloadArray = [];
+        const noOfBatches = Math.ceil(payloadSize / maxPayloadSize);
+        const revisedRecordsPerPayload = Math.floor(payload.data.length / noOfBatches);
+        const revisedDataArray = returnArrayOfSubarrays(payload.data, revisedRecordsPerPayload);
```

Comment:
Can we use `loadash` `chunk` for this

Reply:
No I already checked it can't be used as in output array there may be less data in the last subArray as compared to previous ones

Reply:
So lets say the total size is 10 and revisedRecordsPerPayload = 7
What will be revisedDataArray ?

Reply:
In this case length of subArrays would be [7,3]

Reply:
Thats the same output `chunk` function will return

Reply:
Done

Reply:
Please test out this  cc @shrouti1507

Reply:
I have already generated test Report. Are you concerned about `chunk` function ?

Reply:
yeah, is it working as expected?